src/log.rs

```diff
@@ -125,18 +127,48 @@ unsafe extern "C" fn log_callback(
                 )
             });
             let gobject_type_value = gobject_type_value.as_deref();
-            let gstobject_name = gobject_with_ty.and_then(|(obj, ty)| unsafe {
+            let gstobject = gobject_with_ty.and_then(|(obj, ty)| unsafe {
                 // SAFETY: g_type_is_a is provided valid types.
                 if bool::from_glib(g_type_is_a(ty, gst_object_get_type())) {
                     let gstobject = obj as *mut gstreamer::ffi::GstObject;
-                    // SAFETY: GstObject type has been verified above, `name` can be null and we
-                    // check for it. It "should" be valid null-terminated string if not null,
-                    // however.
-                    Some(CStr::from_ptr((*gstobject).name.as_ref()?).to_string_lossy())
+
+                    Some(gstobject)
                 } else {
                     None
                 }
             });
+
+            let gstobject_name = gstobject.as_ref().and_then(|gstobject| unsafe {
+                // SAFETY: GstObject type has been verified above, `name` can be null and we
+                // check for it. It "should" be valid null-terminated string if not null,
+                // however.
+                Some(CStr::from_ptr((*(*gstobject)).name.as_ref()?).to_string_lossy())
+            });
+
+            let user_span = gstobject.as_ref().and_then(|gstobject| unsafe {
+                let quark = span_quark().into_glib();
+                let mut obj = *gstobject;
+
+                let span = loop {
+                    // lookup object parents until one has a span associated
+                    let span = g_object_get_qdata(obj.cast(), quark);
```

Comment:
Is there a reason you aren’t using [`ObjectExt::qdata`](https://docs.rs/glib/latest/glib/object/trait.ObjectExt.html#tymethod.qdata)?

Reply:
Ah, never mind, I see now – you’re operating on `gstreamer_sys::GstObject`s so that you could grab the `parent`s directly, and using `g_object_get_qdata` is simply more straightforward in this situation…

Though if avoiding a lock on the `gstobject` is a requirement for this code, I worry that we have no way to avoid data between this and `set_qdata` call?

Reply:
> Ah, never mind, I see now – you’re operating on `gstreamer_sys::GstObject`s so that you could grab the `parent`s directly, and using `g_object_get_qdata` is simply more straightforward in this situation…

Yes, exactly.

> Though if avoiding a lock on the `gstobject` is a requirement for this code, I worry that we have no way to avoid data between this and `set_qdata` call?

Let's discuss the locking problem on the other thread so I can close this one.